frontend/src/store/modules/user.ts

```diff
@@ -10,6 +10,9 @@ import _ from "lodash";
 import { Ability } from "@casl/ability";
 import { defineAbilities } from "@/abilities";
 
+let callback: ((snapshot: firebase.database.DataSnapshot) => void) | null;
+let metadataRef: firebase.database.Reference | null;
```

Comment:
I don't like that we have these global variables. Can't we make them module properties?